### ISSUE TITLE
guide: using metrics/plots and params without pipelines

### ISSUE THREAD

**Issue by tupui** [iterative/dvc.org#2216]
UPDATE: Related/part of https://github.com/iterative/dvc.org/issues/2815

My use case involves the inability to really use pipelines. I have models and data, but I am not (or cannot) do the training. So I don't have experiments per se. Still, I have metrics or other metadata I would want to track.

Apparently I can do this using commands like `dvc metrics diff --targets metrics.json`. I think this could be highlighted in the documentation that it is not required to have a dvc.yaml file. Although I would prefer that the `cmd` argument would be optional.

UPDATE: Jump to https://github.com/iterative/dvc.org/issues/2216#issuecomment-781915955

I can do a PR with guidance about where I should put this if wanted. I thought about putting this in a use case (versioning) or in the user guide.

Reference to this proposal is [here](https://discuss.dvc.org/t/how-to-handle-general-metadata-without-experiments/680?u=tupui) . cc @jorgeorpinel

**Comment by shcheklein** [comment]
> Although I would prefer that the cmd argument would be optional.

cc @skshetry

**Comment by jorgeorpinel** [comment]
> prefer that the cmd argument would be optional

Not sure having stages without an action would make sense. But how about accepting `params`, `metrics`, and `plots` in regular .dvc files? They already accept `deps` and `outs` which are the parents of the aforementioned fields.

**Comment by jorgeorpinel** [comment]
> where I should put this if wanted

The command reference would be the first place that comes to mind.

Already mentioned in:
- https://dvc.org/doc/command-reference/params/diff
- https://dvc.org/doc/command-reference/metrics/diff
- https://dvc.org/doc/command-reference/plots/show
- https://dvc.org/doc/command-reference/plots/diff

Not even mentioned yet in:
- https://dvc.org/doc/command-reference/metrics/show

Not supported in:
- https://dvc.org/doc/command-reference/plots/modify

One strategy would be to make sure the mentions are emphatic enough e.g. instead of parenthesis or grayed out notes, make a paragraph (starting with the 💡 emoji perhaps). Also review that it's mentioned in the `--targets` options for commands that have those.

Another idea would be to just leave minimal mentions (double check they're consistent), but emphasize in the **Options** section of each ref: for `targets` argument create an entry that fully describes the behavior and accepted values, like in https://dvc.org/doc/command-reference/repro#options. For `--targets` options similarly expand as much as needed.

**Comment by tupui** [comment]
> > prefer that the cmd argument would be optional
> 
> Not sure having stages without an action would make sense. But how about accepting `params`, `metrics`, and `plots` in regular .dvc files? They already accept `deps` and `outs` which are the parents of the aforementioned fields.

But the value of the main yaml is that the information is centralised. If you just want to give general metadata, there should be an easy way to do so without needing to have stages or anything else. A top level option in the yaml would be better IMO.

**Comment by tupui** [comment]
> One strategy would be to make sure the mentions are emphatic enough e.g. instead of parenthesis or grayed out notes, make a paragraph (starting with the 💡 emoji perhaps). Also review that it's mentioned in the `--targets` options for commands that have those.
> 
> Another idea would be to just leave minimal mentions (double check they're consistent), but emphasize in the **Options** section of each ref: for `targets` argument create an entry that fully describes the behavior and accepted values, like in https://dvc.org/doc/command-reference/repro#options. For `--targets` options similarly expand as much as needed.

Reading at the metrics one for instance, I think the explanation is OK. What I was missing is the fact that I could use all these function without the need to have the yaml file. So in `Project structure` that could maybe be highlighted. Currently it is saying:
```
dvc.yaml pipelines files define stages that form the pipeline(s) of a project. All stage-based features such as dvc params, dvc metrics, and dvc plots are specified here.
```
Which can lead to the conclusion that you have to use this yaml in order to use these functionalities.

**Comment by jorgeorpinel** [comment]
> If you just want to give general metadata, there should be an easy way to do so without needing to have stages or anything else. A top level option in the yaml would be better IMO.

There's a separate discussion around that. See https://github.com/iterative/dvc/issues/3936#issuecomment-692470880
For now file metadata needs it's own .dvc file so I think that's where params/metrics support could be added. If/when tracking data gets incorporated to dvc.yaml that would carry over too.

**Comment by jorgeorpinel** [comment]
BTW that part of the discussion should be in the iterative/dvc repo to be considered. Let's see if @skshetry has no objection on operning that issue (thanks).

**Comment by jorgeorpinel** [comment]
> the metrics one for instance, I think the explanation is OK

Which one? It's not even mentioned in https://dvc.org/doc/command-reference/metrics or https://dvc.org/doc/command-reference/metrics/show.

> in Project structure that could maybe be highlighted

Feel free to send a proposed change (via a PR). Thanks

**Comment by tupui** [comment]
> > the metrics one for instance, I think the explanation is OK
> 
> 
> 
> Which one? It's not even mentioned in https://dvc.org/doc/command-reference/metrics or https://dvc.org/doc/command-reference/metrics/show.
> 

The one for diff. But true it is missing from the general metric discussion.

> 
> > in Project structure that could maybe be highlighted
> 
> 
> 
> Feel free to send a proposed change (via a PR). Thanks

Ok I will try to propose something 😃

**Comment by skshetry** [comment]
Btw, `.dvc` file does support specifying metrics, params and plots. This is a pre-1.0 compatibility feature though. 

We have also plans for supports different sections in `dvc.yaml` file. Eg:
```yaml
metrics:
  - metrics.json
stages:
  ...
 ```

**Comment by jorgeorpinel** [comment]
Merged into #2815.

**Comment by jorgeorpinel** [comment]
#2572, rather.
